### PR TITLE
add scRNA data

### PR DESCRIPTION
**Pull request by armaan-abraham**
- Add scRNA import functions, but not the data itself. Each function takes in path arguments specifying where the data is, with defaults set to the respective locations on Aretha. I think it makes sense to separate the scRNA data from their analysis algorithms.
- Add ruff checks.

### CODE REVIEW

File: poetry.lock [hunk 1]
```diff
@@ -1,33 +1,49 @@
-# This file is automatically @generated by Poetry 1.8.2 and should not be changed by hand.
+# This file is automatically @generated by Poetry 1.8.3 and should not be changed by hand.
```
Comment (aarmey) [inline poetry.lock:1]: Since this will be a dependency to other projects, I'd leave out the lock file from the repo. It doesn't need to be pinned to provide replicable results.

File: tensordata/scRNA.py [hunk 2]
```diff
@@ -0,0 +1,272 @@
+from concurrent.futures import ProcessPoolExecutor
+from pathlib import Path
+
+import anndata
+import numpy as np
+import numpy.typing as npt
+import pandas as pd
+import scanpy as sc
```
Comment (aarmey) [inline tensordata/scRNA.py:8]: Since this will be imported in a lot of places, it would be really, really beneficial to remove the scanpy dependency. Scanpy imports a ton of random packages that makes it fragile and heavy. I think we only need `read_10x_mtx()` so perhaps that code can be copied in here?

File: tensordata/scRNA.py [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+from scipy.sparse import csr_matrix, spmatrix
+from sklearn.utils.sparsefuncs import inplace_column_scale, mean_variance_axis
+
+DATA_DIR = Path(__file__).parent / "scRNA"
+
+
+def prepare_dataset(
```
Comment (aarmey) [inline tensordata/scRNA.py:15]: I'd argue that the normalization is a part of the analysis, and should go in with Pf2.

**Review by aarmey — CHANGES_REQUESTED** [review]
Good idea! Some adjustments here.

File: tensordata/scRNA.py (hunk 3 above)
Comment (armaan-abraham) [reply tensordata/scRNA.py]: Yeah, this also removes the dependence on sc.pp.normalize_total.

File: tensordata/scRNA.py (hunk 2 above)
Comment (armaan-abraham) [reply tensordata/scRNA.py]: This is done, although I removed some options from the scanpy implementation. If we end up needing these options, I can always add them back.

**Comment by armaan-abraham** [comment]
@aarmey This is ready whenever you are!
